### DISCUSSION TITLE
Fine-tuned YOLOv9 models temporarily disabled for Frigate+

### DISCUSSION THREAD

**Discussion by blakeblackshear** [blakeblackshear/frigate#20926] [General]
I am investigating an issue affecting accuracy for some users with YOLOv9 models. I will reach out to affected users directly.

This issue does not impact the base model, so I recommend using the base YOLOv9 model or fine tuning a YOLO-NAS model instead.

**Comment by xconverge** [comment]
Take your time to investigate of course, but a question I have if/when answerable is does/did this impact all variants of yolov9 fine tuned (like hailo8l)? 

I will just wait for any updates related to this and subscribe to this discussion

I am curious because I feel like I HAD somewhat of a rocky start from my switch from mobiledet to yolov9, but feel like after quite a few fine tunings, I got the accuracy and improvements I was looking for in the end and seem to be problem free now, it just took more fine tunings and images than I expected

**Reply by blakeblackshear** [reply]
Yes. It would impact all variants. I'm not surprised by your rocky start. The good news is that the additional training data it required will just make it even better after the changes I am working on. The issues would have been far more prevalent for users that had smaller numbers of images or certain labels with few/no examples. This is why it wasn't obvious to myself or any of the maintainers since we each have large image counts built over a long period of time.

**Reply by Fratopolis** [reply]
Well I'll Be.........I Have been training like crazy since the realease of v9 and buying more training adding thousands and thousands of images and even deleting several hundred images thinking some were causing issues. Even went through every single already submitted image manually.  Now I wish I would have mentioned something. Would have saved a little moola and maybe caught this earlier for everyone. 

LOL price we pay if we want bleeding edge with hiccups. Still money well spent to support the project.

BTW I went to a "Brand I shall not name publicly (blake let me know if curious and I can elaborate more in an email to you)" confernce for work recently and they were showing off their latest VMS solution. One of the reps was talking to me and I showed them my home Frigate install and lets just say she was very interested or more so blown away it seemed and wanted to know what product I was using.

**Comment by blakeblackshear** [comment]
I released an update. Here is the announcement I posted on the website for those following this discussion.

> After investigating reports of accuracy issues with YOLOv9 models, I discovered an issue with the parameters used during fine tuning. Specifically, the learning rate was set too high, which caused the models to overfit to your data and forget a significant portion of the learning from the training before fine tuning. This was particularly problematic for users looking to detect objects with fewer examples in their own data.
> 
> I conducted several experiments and updated the parameters used during fine tuning to more optimal values. As a result, fine tuned models should now exhibit improved accuracy and reliability. Users who have large, well annotated datasets may not see a noticeable improvement, but those with smaller datasets should benefit significantly.

**Reply by xconverge** [reply]
Thanks @blakeblackshear! Sounds logical, super appreciate the update!

~Is there a way to see `Trainings remaining: 7` the breakdown here of ones that are part of a subscription (that will go away on a certain date potentially) vs ones that will persist because were bought after the subscription ones were used?~

Edit: I found it in the API request to `https://api.frigate.video/v1/user/profile`
